src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java

```diff
@@ -0,0 +1,9 @@
+package uk.gov.register.core;
+
+import uk.gov.verifiablelog.store.memoization.MemoizationStore;
+
+public interface TransactionalMemoizationStore extends MemoizationStore {
+    void commitEntries();
+    void rollbackEntries();
+    void setCurrentEntryCount(int entryCutoff);
```

Comment:
- different param name `currentEntryCount` in the impl
  - any reason the field's type is `Integer` but the param `int`?